src/javaRosa.js

```diff
@@ -1055,6 +1054,85 @@ define([
 
     };
 
+    var itextMarkdownWidget = function (mug, language, form, options) {
+        options = options || {};
+        var parent = options.parent;
+        var widget = itextLabelWidget(mug, language, form, options),
+            super_setValue = widget.setValue,
+            super_getUIElement = widget.getUIElement,
+            super_handleChange = widget.handleChange,
+            wantsMarkdown = true,
+            markdownOff, markdownOn, markdownOutput;
+
+        widget.toggleMarkdown = function() {
+            parent.toggleClass("has-markdown");
+        };
+
+        markdownOutput = $('<div>').addClass("controls well markdown-output");
+
+        widget.handleChange = function() {
+            super_handleChange();
+            var val = widget.getValue(),
+                item = this.getItextItem();
+            if (/[~*#[\]]+|^-/m.test(val)) {
+                if (wantsMarkdown) {
+                    parent.removeClass("markdown-ignorant");
+                    parent.addClass("has-markdown");
+                }
+            } else if (!val) {
+                parent.removeClass("has-markdown");
+            }
+            item.hasMarkdown = markdownOff.is(":visible");
+            markdownOutput.html(util.markdownFull(val)).removeClass('hide');
+        };
+
+        widget.setValue = function (val) {
+            super_setValue(val);
+            if (!val) {
+                markdownOutput.addClass('hide');
+            }
+            markdownOutput.html(util.markdownFull(val));
+        };
+
+        widget.getUIElement = function() {
+            var elem = super_getUIElement(),
+                val = widget.getValue();
+
+            elem.detach('.markdown-output');
+            elem.append(markdownOutput);
+            elem.find('.control-label').append(markdown_help({title:options.lstring }));
+
+            markdownOff = elem.find('.turn-markdown-off').click(function() {
+                wantsMarkdown = false;
+                var item = widget.getItextItem();
+                item.hasMarkdown = false;
```

Comment:
stylistically it seems cleaner to just combine these two items and get rid of the variable.